src/main/java/de/tum/in/www1/artemis/service/metis/ConversationMessagingService.java

```diff
@@ -0,0 +1,232 @@
+     * @return the created message
+     */
+    public Post createMessage(Long courseId, Post newMessage) {
+        if (newMessage.getId() != null) {
+            throw new BadRequestAlertException("A new message post cannot already have an ID", METIS_POST_ENTITY_NAME, "idexists");
+        }
+        if (newMessage.getConversation() == null || newMessage.getConversation().getId() == null) {
+            throw new BadRequestAlertException("A new message post must have a conversation", METIS_POST_ENTITY_NAME, "conversationnotset");
+        }
+
+        var author = this.userRepository.getUserWithGroupsAndAuthorities();
+        var course = preCheckUserAndCourse(author, courseId);
+        newMessage.setAuthor(author);
+        newMessage.setDisplayPriority(DisplayPriority.NONE);
+
+        var conversation = conversationService.mayInteractWithConversationElseThrow(newMessage.getConversation().getId(), author);
+        // extra checks for channels
+        if (conversation instanceof Channel channel) {
+            channelAuthorizationService.isAllowedToCreateNewPostInChannel(channel, author);
+        }
+
+        // update last message date of conversation
+        conversation.setLastMessageDate(ZonedDateTime.now());
+        conversation = conversationService.updateConversation(conversation);
+        // update last read date and unread message count of author
+        var authorParticipant = conversationParticipantRepository.findConversationParticipantByConversationIdAndUserIdElseThrow(conversation.getId(), author.getId());
+        authorParticipant.setLastRead(ZonedDateTime.now());
+        authorParticipant.setUnreadMessagesCount(0L);
+        conversationParticipantRepository.save(authorParticipant);
+
+        var createdMessage = conversationMessageRepository.save(newMessage);
+        broadcastForPost(new PostDTO(createdMessage, MetisCrudAction.CREATE), course);
+
+        if (conversation instanceof OneToOneChat) {
+            var getNumberOfPosts = conversationMessageRepository.countByConversationId(conversation.getId());
+            if (getNumberOfPosts == 1) { // first message in one to one chat --> notify all participants that a conversation with them has been created
+                var participants = conversationParticipantRepository.findConversationParticipantByConversationId(conversation.getId()).stream()
+                        .map(ConversationParticipant::getUser).filter(Objects::nonNull).collect(Collectors.toSet());
+                conversationService.broadcastOnConversationMembershipChannel(course, MetisCrudAction.CREATE, conversation, participants);
+            }
+        }
+        conversationParticipantRepository.incrementUnreadMessagesCountOfParticipants(conversation.getId(), author.getId());
+        // ToDo: Optimization Idea: Maybe we can save this websocket call and instead get the last message date from the conversation object in the post somehow?
```

Comment:
Open todo.

Reply:
Will be done in a follow up. I sometimes write todos for optimization ideas